nncf/quantization/algorithms/bias_correction/algorithm.py

```diff
@@ -472,16 +494,49 @@ def _get_biased_after_input_nodes(self, nncf_graph: NNCFGraph, model_inputs: Lis
         :return: A dictionary with the names of the nodes with bias as keys and their input node names as values.
         """
         def traverse_to_biased(node, output):
-            if self._backend_entity.is_node_with_bias(node):
+            if node in output:
+                return True, output
+            if self._backend_entity.is_node_with_bias(node, nncf_graph):
                 output.append(node)
                 return True, output
             return False, output
 
         biased_after_param_nodes = {}
 
         for model_input in model_inputs:
-            biased_nodes = nncf_graph.traverse_graph(model_input, traverse_to_biased)
-            for biased_node in biased_nodes:
-                activation_input = self._backend_entity.get_node_through_quantizer(biased_node, nncf_graph)
-                biased_after_param_nodes[biased_node.node_name] = activation_input.node_name
+            nodes_with_bias = nncf_graph.traverse_graph(model_input, traverse_to_biased)
+            for node in nodes_with_bias:
+                activation_input = self._get_previous_activation_node(node, nncf_graph)
+                biased_after_param_nodes[node.node_name] = activation_input.node_name
         return biased_after_param_nodes
+
+    def _get_previous_activation_node(self, node: NNCFNode, nncf_graph: NNCFGraph) -> NNCFNode:
+        """
+        Returns 0- activation node, but not quanitzers.
+
+        :param node: NNCFNode instance.
+        :param nncf_graph: NNCFGraph instance.
+        :return: NNCFNode activation node.
+        """
+        activation_input_port = 0
+        previous_node = nncf_graph.get_input_edges(node)[activation_input_port].from_node
+        return previous_node
```

Comment:
I don't see the reason why this method is necessary. The docstring is not very clear to me.
My opinion is it is better to use
```python
node = nncf_graph.get_input_edges(node)[0].from_node
```
and add comments for this line to give more information if it is needed.